### PR TITLE
Stub out API 

### PR DESCRIPTION
**Pull request by chadlagore**
:hourglass: **Status**: _Ready for review_
:tickets: **Ticket(s)**:

## :construction_worker: Changes

+ Added `gunicorn` for hot reload.
+ Removed dev environment, two environments is not the right pattern for this project.
+ Added a light auth decorator. TODO: Test it.

## :flashlight: Testing Instructions

```
make
make run
```

Meanwhile...

```bash
curl -sSL -X POST 0.0.0.0/events/1/rsvp   # Error: rsvp_handler not implemented%
curl -sSL -X DELETE 0.0.0.0/events/1/rsvp   # Error: rsvp_handler not implemented%
curl -sSL -X GET 0.0.0.0/users/self  # Error: get_self not implemented%
curl -sSL -X GET 0.0.0.0/   # {"hello":"async world"}%
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,10 +5,15 @@ ADD requirements.txt /env/requirements.txt
...
 WORKDIR /app/
 
-ENTRYPOINT [ "python3" ]
-
-CMD [ "beluga/app.py" ]
+# Hot reload on app from code modifications.
+CMD gunicorn \
+    --reload \
+    --bind 0.0.0.0:80 \
+    --workers 4 \
```
Comment (agentfll) [inline Dockerfile]: I think these should vary based on env, maybe implementable with a simple bash script. In particular, I think we should turn off reloading in production, have one (maybe two, depending on concurrency properties of the program) worker in dev, and bind to 127.0.0.1 in dev (for security, so only we can access our dev server by default)

Comment (chadlagore) [reply Dockerfile]: Agreed here. I like the bash script entrypoint approach.

**Comment by chadlagore** [comment]
As for the dev image, I pulled it out because I felt pytest and apidoc were minor overheads to keep our development cycle a bit simpler. The dev image itself doesnt actually get shipped anywhere (in other projects I'm doing that). We can bring it back if you think you'll use it. Since the folder is mounted, you basically are always editing code inside the container.

**Comment by agentfll** [comment]
@chadlagore sounds good on dropping dev for now--I reckon we'll want a way to build and push a prod image at some point, but we might be able to do so by taking a dev image and removing dev dependencies and modifying the env without touching the actual code (i.e. a "promote artifact" `make` target)

File: bin/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        --bind 0.0.0.0:80 \
+        --workers 4 \
+        --worker-class \
+        sanic_gunicorn.Worker \
+        beluga:app
+else
+    echo "Running for development."
+    gunicorn \
+        --reload \
+        --bind 127.0.0.1:80 \
```
Comment (agentfll) [inline bin/entrypoint.sh]: Mismatch w/ `DEV_PORT`

File: Makefile [hunk 3]
```diff
@@ -14,15 +14,24 @@ PROD_REMOTE=$(ORGANIZATION)/$(PROD_LOCAL)
...
 all: build-prod
 
-run:
+run-dev:
 	docker rm -f $(PROD_CONTAINER) 2>> /dev/null || true
 	docker run -it \
 		--name $(PROD_IMAGE_NAME) \
-		-v `pwd`/beluga:/app/beluga \
+		-v `pwd`:/app \
+		-p $(PROD_PORT):$(PROD_PORT) \
```
Comment (agentfll) [inline Makefile]: `DEV_PORT`?

File: beluga/routes.py [hunk 4]
```diff
@@ -14,33 +14,69 @@
...
+    @apiSuccess {String} next URL of the next page of results.
+    @apiSuccess {String} previous URL of the previous page of results.
+    """
     raise NotFound('event_handler not implemented')
 
 
 @authorized()
 async def rsvp_handler(request, uuid):
+    """
+    @api {get} /events/:uuid/rsvp RSVP to an event (i.e. become an attendee).
```
Comment (agentfll) [inline beluga/routes.py]: Missing DELETE method

File: beluga/routes.py [hunk 5]
```diff
@@ -14,33 +14,69 @@
...
     Sets the authorized user's RSVP status
     on the event specified by `uuid` to False.
     """
     pass
 
 
 async def post_rsvp(token, uuid):
-    """
-    DELETE RSVP handler.
+    """DELETE RSVP handler.
```
Comment (agentfll) [inline beluga/routes.py]: Error in comment

**Comment by chadlagore** [comment]
I know where they live, kind of.

**Pull request merged**